pkg/operator/ceph/cluster/mgr/dashboard_sso.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright (c) 2023 Koor Technolgies, Inc.
+*/
+// Package mgr for the Ceph manager.
+package mgr
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/util"
+	"github.com/rook/rook/pkg/util/exec"
+)
+
+const (
+	dashboardUserReadOnlyRole = "read-only"
+)
+
+type DashboardSSOInfo struct {
+	OneLoginSettings OneLoginSettings `json:"onelogin_settings"`
+}
+
+type OneLoginSettings struct {
+	// TODO Check which fields are important to check against the CRD input
+}
+
+func (c *Cluster) configureSSO() (bool, error) {
+	if !c.spec.Dashboard.SSO.Enabled {
+		// Check if SSO is still enabled
+		args := []string{"dashboard", "sso", "status"}
+		output, err := client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
+			output, err := client.NewCephCommand(c.context, c.clusterInfo, args).RunWithTimeout(exec.CephCommandsTimeout)
+			return "checking dashboard sso status", output, err
+		}, c.exitCode, 5, invalidArgErrorCode, dashboardInitWaitTime)
```

Comment:
The 4th argument is the code we retry on. `invalidArgErrorCode` is different than what it was when it was a loop... Is this intended?

Reply:
hey @galexrt can you help in explaining this bit here ? I believe we want to retry on this error code intentionally here if i understand correctly.

Reply:
The previous checked if the command timed out and continued the loop, the `invalidArgErrorCode` is "standard" as if the dashboard module is not enabled/still starting, that error is returned for most/any dashboard-related commands.

So this makes it a little bit cleaner in regards to handling that scenario.